sdks/python/apache_beam/transforms/resources.py

```diff
@@ -134,6 +134,8 @@ def _parse_storage_size_str(value):
       raise ValueError()
 
     suffix = match.group(1)
+    if suffix not in units:
+      raise ValueError()
```

Comment:
Let's put a more informative message here (e.g. ValueError("Unknown unit: %s" % suffix). Similarly above. 

Also, do we want suffixes to be case sensitive. E.g. should we be rejecting "4gb"?

Reply:
there is one interpretation where lowercase b means bits.. so wanted to avoid this confusion just in case even though unlikely
https://en.wikipedia.org/wiki/JEDEC_memory_standards#Unit_prefixes_for_semiconductor_storage_capacity

Reply:
Proper error will be generated in https://github.com/tvalentyn/beam/blob/3d01befb49d5c213e62e44aaccca76c4b809ea61/sdks/python/apache_beam/transforms/resources.py#L184

The keyerror was confusing that logic.

Reply:
OK. I suppose we can always loosen it up later more easily than tighten it up. 

Even though errors are caught and re-raised, I prefer messages here, but will leave this up to you.

Reply:
Ah, I see, sure, that wouldn't hurt and is more readable.

Reply:
Done.